backend/tests/main_test.py

```diff
@@ -18,28 +23,33 @@ def test_message_users():
     assert response.json() == {"message": "Hello User!!!"}
 
 
-def test_user_session_flow():
-    # Client logs in.
-    response = client.post(
-        "api/users/login",
-    )
-    session_cookie = response.cookies.get_dict()
-    assert response.status_code == 200
-    assert session_cookie[settings.server_sessions.session_key] is not None
-    assert response.json() == {"message": "successfully logged in"}
-
-    # Set the 'matrix_user' field in session database
-    name = "John Paul"
-    params = {"name": name}
-    response = client.post("api/users/changeTokens", params=params)
-
-    assert response.status_code == 200
-    assert response.json() == {"message": "success"}
-
-    # fetch the 'matrix_user' field in session database
+@pytest.mark.asyncio
+async def test_user_session_flow():
+    matrix_user = "@example_user:matrix.org"
+    with aioresponses() as mock_server:
```

Comment:
It's probably a little easier to read (avoids the context manager indentation) if you put this in a fixture (from the aioresponses readme):

```python import pytest
from aioresponses import aioresponses

@pytest.fixture
def mock_aioresponse():
    with aioresponses() as m:
        yield m
```